applications/GeoMechanicsApplication/custom_conditions/U_Pw_normal_lysmer_absorbing_condition.hpp

```diff
@@ -98,23 +131,62 @@ class KRATOS_API(GEO_MECHANICS_APPLICATION) UPwLysmerAbsorbingCondition : public
 
-    void CalculateNodalDampingMatrix(NormalLysmerAbsorbingVariables& rVariables, const ProcessInfo& CurrentProcessInfo, const Element::GeometryType& Geom);
-
-    void CalculateNodalStiffnessMatrix(NormalLysmerAbsorbingVariables& rVariables, const ProcessInfo& CurrentProcessInfo, const Element::GeometryType& Geom);
-
-    Matrix CalculateExtrapolationMatrixNeighbour(const Element& NeighbourElement);
+    /**
+    * @brief Adds the condition matrix to the global left hand side
+    * @param rLeftHandSideMatrix Global Left hand side
+    * @param rUMatrix LHS displacement matrix of the current condition
+    */
+    void AddLHS(MatrixType& rLeftHandSideMatrix, const BoundedMatrix<double, N_DOF, N_DOF>& rUMatrix);
+
+    /**
+    * @brief Calculates and adds terms to the RHS
+    * @param rRigtHandSideMatrix Global Right hand side
+    * @param rStiffnessMatrix condition stiffness matrix
+    */
+    void CalculateAndAddRHS(VectorType& rRightHandSideMatrix, const MatrixType& rStiffnessMatrix);
+
+    /**
+    * @brief Calculates the rotation matrix of the current condition
+    * @param rRotationMatrix rotation matrix of the current condition
+    * @param rGeom geometry of the current condition
+    */
+    void CalculateRotationMatrix(BoundedMatrix<double, TDim, TDim>& rRotationMatrix, const Element::GeometryType& rGeom);
+
+    /**
+    * @brief This method gets the average of the variables of all the neighbour elements of the condition. 
+    * @param rVariables Condition specific variables struct
+    * @param rCurrentProcessInfo Current process information
+    */
+    void GetNeighbourElementVariables(NormalLysmerAbsorbingVariables& rVariables, const ProcessInfo& rCurrentProcessInfo);
+
+    /**
+    * @brief Gets condition variables.
+    * @param rVariables Condition specific variables struct
+    * @param rCurrentProcessInfo Current process information
+    */
+    void GetVariables(NormalLysmerAbsorbingVariables& rVariables, const ProcessInfo& rCurrentProcessInfo);
+
+    /**
+     * @brief Calculates the damping constant in all directions
+     * @param rVariables Condition specific variables struct
+     * @param rGeom condition geometry
+     */
+    void CalculateNodalDampingMatrix(NormalLysmerAbsorbingVariables& rVariables, const Element::GeometryType& rGeom);
+
+    /**
+    * @brief Calculates the stiffness constant in all directions
+    * @param rVariables Condition specific variables struct
+    * @param rGeom condition geometry
+    */
+    void CalculateNodalStiffnessMatrix(NormalLysmerAbsorbingVariables& rVariables, const Element::GeometryType& rGeom);
+
+    /**
+    * @brief Calculates the extrapolation matrix for neighbour elements.Values from integration points are extrapolated to the nodes
+    * @param rNeighbourElement The neighbouring element of the condition
+    */
+    Matrix CalculateExtrapolationMatrixNeighbour(const Element& rNeighbourElement);
```

Comment:
This is for me the preferred way of returning results (using the return type, rather than by modifying one of the function's arguments as done above). I have noticed that for some cases we are forced to return the result through the argument list (e.g. when a member function is overridden). However, when we have the freedom to decide for ourselves, I'd prefer to use the return type for that.